content/en/blog/2024/otel-generative-ai/index.md

```diff
@@ -0,0 +1,147 @@
+---
+title: OpenTelemetry for Generative AI
+linkTitle: OpenTelemetry for Generative AI
+date: 2024-11-09
+author: >-
+  [Drew Robbins](https://github.com/drewby) (Microsoft),  [Liudmila
+  Molkova](https://github.com/lmolkova) (Microsoft)
+---
+
+As organizations increasingly adopt Large Language Models (LLMs) and other
+generative AI technologies, ensuring reliable performance, efficiency, and
+safety is essential to meet user expectations, optimize resource costs, and
+safeguard against unintended outputs. Effective observability for AI operations,
+behaviors, and outcomes can help meet these goals. OpenTelemetry is being
+enhanced to support these needs specifically for generative AI.
+
+Two primary assets are in development to make this possible: **Semantic
+Conventions** and an **Instrumentation Library**.
```

Comment:
```suggestion
Conventions** and an **Instrumentation Libraries**.
```

Is there only one library that is developed or are there multiple libraries?

Reply:
There is only one at this moment. We focused on Python first.

Reply:
then maybe write this here already:

```suggestion
Conventions** and an **Instrumentation Library** for OpenAI's APIs:
```

Reply:
If we made this change, it would conflate the GenAI working group assets with OpenAI, which I think doesn't follow later. Also, it feels less attractive to folks who are not OpenAI based, yet should be welcome into the team. Also, the specs are used for non-openai right now. It is just that the first library instrumentation in the shared codebase is openai.

If we want to hint what's first, I think we should make that a second sentence. wdyt?

```suggestion
Conventions** and an **Instrumentation Library**. The first instrumentation library targets OpenAI in Python.
```